csm/SystemMan.py

```diff
@@ -0,0 +1,139 @@
+import CommsMan as CommsMan
+import StorageMan as StorageMan
+import MotorMan as MotorMan
+import CameraMan as CameraMan
+import PerceptionMan as PerceptionMan
+
+from CommsMan import CommsMan
+from StorageMan import StorageMan
+from MotorMan import MotorMan
+from CameraMan import CameraMan
+from PerceptionMan import PerceptionMan
+
+# from ImageSources import ImageSource, LocalVideo, StillImage, Camera
+# from PerceptionUtils import BoundingBox
+
+import queue
+import threading
+import time
+import sys
+
+class SystemMan():
+
+    def __init__(self):
+        self.running = True
+        self.inFrame = False
+        self.currVideo = 0
+
+        # Instantiate Subsystems
+        self.per = PerceptionMan(threshold=0.5)
+        self.cam = CameraMan("../testData/testOutput.mp4")
+        self.mot = MotorMan()
+        
+
+        self.sto = StorageMan()
+        self.comQ, self.comT = queue.Queue(maxsize=1), queue.Queue(maxsize=1)
+        self.com = CommsMan(self.comQ, self.comT)
+
+    def sendMessageToRemote(self, message):
+        self.com.sendMessageToRemote(message)
+
+    def terminateCommsMan(self):
+        return self.com.terminateCommsMan()
+
+    def simulateReceiveBT(self, message):
+        self.com.simulateReceiveBT(message)
+
+    def simulateLogBluetooth(self):
+        return self.com.simulateLogBluetooth()
+
+    def SHUTDOWN(self):
+        self.simulateReceiveBT("Terminate")
+
+    def launch(self):
+        # Instantiate and launch in threads: CommsMan, StorageMan
+        comThread = threading.Thread(target=(self.com.launch))
+        stoThread = threading.Thread(target=(self.sto.launch))
+        comThread.start()
+        stoThread.start()
+
+        while self.running:
+
+            if not self.comQ.empty():
+                # Pull and process message from CommsMan
+                msg = self.comQ.get()
+
+                print("Main         : received message: %s" % msg)
+
+                # If msg commands new target selection
+                if (msg == "Terminate"):
+                    break
+                elif (msg != ''):
+                    # Restart PerceptionMan.
```

Comment:
Could you add explicit TODOs here for what's missing?

Reply:
Todos refer to portions of code you previous included in commit ec07fc48b9f7a76fe7cacf7b985482906d25aa21 - when you refactored ImageSources. How specific do you need the TODOs to be?

Reply:
Oh I just meant something like "# TODO: Call perception function that does this or whatever", I'll get rid of them once I do it so nothing fancy